### ISSUE TITLE
[Feature request] Winget publication

### ISSUE THREAD

**Issue by NicolasConstant** [beeradmoore/dlss-swapper#159]
**Is your feature request related to a problem? Please describe.**
Would be nice to have dlss-swapper published automatically on winget's depo

**Describe the solution you'd like**
Adding a github action to create automatically a PR on winget's depo for the new version/build of dlss-swapper

**Additional context**
To help you getting started, I've submitted the [initial profile on winget](https://github.com/microsoft/winget-pkgs/pull/119914/files), it's already available and usable, check it with:
```
winget search dlss-swapper
winget install dlss-swapper
``` 
You could just fork and update the profile for the next releases, or looking to those fancy github action to do it automatically.

I hope the little kickstart will help!

**Comment by beeradmoore** [comment]
Hey @NicolasConstant , thanks for doing that.

I had tried this a few times in the past. First time was when we were still a self signed msix which required certificate bundle which wouldn't work. Forget what happened the second time but it was another constraint with it.

Glad it is up and able to work now. Thanks for adding the initial work for it. I'll see what I can do with automation, if not at minimum will make sure I have a release checklist to make sure it's added.

**Comment by beeradmoore** [comment]
Just re-installed my PC and was great to be able to install with winget :)

I was thinking more how to integrate this into some pipeline or something. What I think I'll do is make another tool in the the package folder for building winget manifests (and maybe pull + create PR automation). That way it can read from the GitHub release notes to build everything else. I'll then write correct docs on the publishing flow so I can ensure I follow it to be sure winget is updated.

I'll also double check the behaviour of `UpgradeBehavior: uninstallPrevious`. It's another thing I hadn't really thought of which is users updating DLSS Swapper with the installer, should it uninstall, should it overwrite, etc. With that I'll also need to check what it is going to do with downloaded DLSS packages and other user configs as it may nuke them both.

**Comment by NicolasConstant** [comment]
If it can help, on my own software, I'm planning using this [github action](https://github.com/marketplace/actions/winget-releaser) for the automation part. It's based on the software I used to generate the first profile, and it seems straightforward.

**Comment by beeradmoore** [comment]
I have just double checked. Installing DLSS Swapper, downloading dlss, uninstalling does not wipe the stored data. I am going to file that as an issue to add to the uninstaller but going to not have it do it by default so updating with `UpgradeBehavior: uninstallPrevious` still works fine.

Thanks for that GitHub action. When I read it a few days ago I thought I'd have to change building to be on GitHub actions (which is the plan in the future), but thankfully it isn't and this looks A LOT easier to implement. I'll look at doing that once @SpecterShell has done any little tweaks they'd like to do.

**Comment by beeradmoore** [comment]
I just release v1.0.4, I'll look at updating winget to install this version over the next few days.

**Comment by beeradmoore** [comment]
Narrator: He did not get that to winget within a few days

Side note, I am putting this here to be able to automate this when publishing packages so winget will be up to date always.
https://learn.microsoft.com/en-us/shows/open-at-microsoft/wingetcreate-keeping-winget-packages-up-to-date

**Comment by beeradmoore** [comment]
I have tested wingetcreate locally to submit an update to winget which looks like its on track to working for v1.0.5. I put this into a github [action](https://github.com/beeradmoore/dlss-swapper/blob/main/.github/workflows/winget-submission.yml) so this should now handle future publishes automatically.

Next steps will be to move entire build process into github actions, not just the winget publishing.

Closing this for now now as it is complete. Thanks again for doing all the legwork to get this started.
